### ISSUE TITLE
Request: Support for conditions in location fields (Waypoint, Destination, Source)

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#5525]
**Is your feature request related to a problem? Please describe.**
I am currently trying to program a job so that it will exclude systems with a particular attribute until after a specific mission is complete.

```
	waypoint
		not attributes "pleiades"
		or
			not attribute "cognizance 1"
			and
				attribute "cognizance 1"
				"Remnant: Cognizance 18: done"
```
This should make it so that it will:
- Never pick a system with attribute "pleiades"
- Never pick a system with attribute "cognizance 1" unless the pilot has "Remnant: Cognizance 18: done"

**Related Issue Links**
Not that I could see. 

**Describe the solution you'd like**
Implementation of conditions including `and` and `or` in the same manner as currently used for missions.

**Describe alternatives you've considered**
Duplicating the job and having one apply before Cognizance 18, and another apply afterwards. 

**Additional context**
This currently came up because the Astral research job sent someone to a system that wasn't accessible yet because they hadn't done the Cognizance missions, and reported it as a bug on Discord by PerspectiveBleed (Fu4ow).

**Comment by oo13** [comment]
I agree ```and``` and ```or``` are usefull but we can work around it.

Use De Morgan's law.
```
	waypoint
		not
			attributes "pleiades"
			not
				attributes "cognizance 1"
				not
					attributes "cognizance 1"
					"Remnant: Cognizance 18: done"
```
... Location filter does not support ```"Remnant: Cognizance 18: done"```.
